tests/core/full_node/stores/test_coin_store.py

```diff
@@ -493,6 +495,183 @@ async def test_get_coin_states(db_version: int) -> None:
         assert len(await coin_store.get_coin_states_by_ids(True, coins, uint32(0), max_items=10000)) == 600
 
 
+# Generate coin records, puzzle hashes, and hints.
+batch_coin_records: List[CoinRecord] = []
+batch_puzzle_hashes: List[bytes32] = []
+batch_hints: List[Tuple[bytes32, bytes]] = []
+
+for i in range(50000):
+    is_spent = i % 2 == 0
+    is_hinted = i % 7 == 0
+    created_height = uint32(i)
+    spent_height = uint32(created_height + 100)
+
+    puzzle_hash = std_hash(i.to_bytes(4, byteorder="big"))
+
+    coin = Coin(
+        std_hash(b"Parent Coin Id " + i.to_bytes(4, byteorder="big")),
+        puzzle_hash,
+        uint64(1000),
+    )
+
+    if is_hinted:
+        hint = std_hash(b"Hinted " + puzzle_hash)
```

Comment:
unless I'm missing something, this hint won't ever match a puzzle hash that you're selecting for in your query. in this test, do you ever pick up *any* coin records because `include_hinted` is `True`?
See comment below

Reply:
I believe it does